### DISCUSSION TITLE
How to use types from upcoming Python version?

### DISCUSSION THREAD

**Discussion by dimaqq** [microsoft/pyright#8126] [Q&A] [answered]
Consider `warnings.deprecated` which typeshed already has type for:

https://github.com/python/typeshed/blob/804a58165a59b88d933bc64b43172b3747ae9957/stdlib/warnings.pyi#L118-L124

But the type is behind `if sys.version_info >= (3, 13):` and I'm planning to vendor a compatible decorator in my library, as it's meant to be used across wider set of Python versions.

I'd love to have `pyright` pick up the type from typeshed.

How would I go about doing that?

**Comment by erictraut** [comment, marked as answer]
You can import `deprecated` from the `typing_extensions` library, which provides backward compatibility for newer typing features.
